src/main/java/org/apache/sysds/runtime/matrix/data/FrameBlock.java

```diff
@@ -2345,6 +2384,77 @@ public FrameBlock replaceOperations(String pattern, String replacement){
 		return ret;
 	}
 
+	public  FrameBlock removeEmptyOperations(boolean rows, boolean emptyReturn, MatrixBlock select) {
+		if( rows )
+			return removeEmptyRows(select, emptyReturn);
+		else //cols
+			return removeEmptyColumns(select, emptyReturn);
+	}
+
+	private FrameBlock removeEmptyRows(MatrixBlock select, boolean emptyReturn) {
+		FrameBlock ret = new FrameBlock(_schema, _colnames);
+
+		for(int i = 0; i < _numRows; i++) {
+			boolean isEmpty = true;
+			Object[] row = new Object[getNumColumns()];
+
+			for(int j = 0; j < getNumColumns(); j++) {
+				Array colData = _coldata[j].clone();
+				row[j] = colData.get(i);
+				ValueType type = _schema[j];
+				isEmpty = isEmpty && (ArrayUtils.contains(new double[]{0.0, Double.NaN}, UtilFunctions.objectToDoubleSafe(type, colData.get(i))));
+			}
+
+			if((!isEmpty && select == null) || (select != null && select.getValue(i, 0) == 1)) {
```

Comment:
Please add a TODO for the remaining functionalities like support for select vector and margin = "cols"